tests/ZohoDatabaseSyncZohoTest.php

```diff
@@ -0,0 +1,117 @@
+        // Test insert
+        $data = [
+            'firstName' => 'TestZohoSync',
+            'lastName' => uniqid('Test'),
+            'dateOfBirth' => date('Y-m-d')
+        ];
+        $data['email'] = $data['lastName'].'@test.com';
+        $this->dbConnection->insert($tableName, $data);
+        $statementTestInsert = $this->dbConnection->createQueryBuilder();
+        $statementTestInsert->select('zcrm.*')
+        ->from('local_insert', 'l')
+        ->join('l', $tableName, 'zcrm', 'zcrm.uid = l.uid')
+        ->where('l.table_name=:table_name')
+        ->setParameters([
+            'table_name' => $tableName
+        ]);
+        $resultInsertion = $statementTestInsert->execute()->fetchAll();
+        $this->assertNotFalse($resultInsertion);
+        $zohoZync->pushInsertedRows($contactZohoDao);
+        $resultContactInserted = $this->dbConnection->fetchAssoc('SELECT * FROM '.$tableName.' WHERE uid = :uid', ['uid' => $resultInsertion[0]['uid']]);
+        $this->assertNotFalse($resultContactInserted);
+        $this->assertNotNull($resultContactInserted['id']);
+        
+        //Test update
+        $dataUpdate = [
+            'firstName' => 'TestZohoSyncUpdated',
+            'lastName' => uniqid('TestUpdated'),
+        ];
+        $this->dbConnection->update($tableName, $dataUpdate, ['id' => $resultContactInserted['id']]);
+        $statementTestUpdate = $this->dbConnection->createQueryBuilder();
+        $statementTestUpdate->select('l.field_name as updated_fieldname')
+        ->from('local_update', 'l')
+        ->join('l', $tableName, 'zcrm', 'zcrm.uid = l.uid')
+        ->where('l.table_name=:table_name')
+        ->andWhere('l.uid = :uid')
+        ->setParameters([
+            'table_name' => $tableName,
+            'uid' => $resultInsertion[0]['uid']
+        ]);
+        $resultUpdate = $statementTestUpdate->execute()->fetchAll();
+        $this->assertNotFalse($resultUpdate);
+        $this->assertCount(2, $resultUpdate);
+        $zohoZync->pushUpdatedRows($contactZohoDao);
+        sleep(60);
+        $contactZohoUpdate = $contactZohoDao->getById($resultContactInserted['id']);
+        $this->assertEquals($dataUpdate['firstName'],$contactZohoUpdate->getFirstName());
+        $this->assertEquals($dataUpdate['lastName'], $contactZohoUpdate->getLastName());
+
+        // Delete
+        $this->dbConnection->delete($tableName, ['uid' => $resultInsertion[0]['uid']]);
+        $statementTestDelete = $this->dbConnection->createQueryBuilder();
+        $statementTestDelete->select('l.*')
+        ->from('local_delete', 'l')
+        ->where('l.table_name=:table_name')
+        ->andWhere('l.id = :id')
+        ->setParameters([
+            'table_name' => $tableName,
+            'id' => $resultContactInserted['id']
+        ]);
+        $resultDelete= $statementTestDelete->execute()->fetchAll();
+        $this->assertNotFalse($resultDelete);
+        $zohoZync->pushDeletedRows($contactZohoDao);
+        
```

Comment:
We should add a test here to try to fetch the data from Zoho by ID and see that it throws an error.